lib/Cake/Model/Behavior/TreeBehavior.php

```diff
@@ -97,11 +97,48 @@ public function afterSave(Model $Model, $created, $options = array()) {
 			}
 		} elseif ($this->settings[$Model->alias]['__parentChange']) {
 			$this->settings[$Model->alias]['__parentChange'] = false;
+			if ($level) {
+				$this->_setChildrenLevel($Model, $Model->id);
+			}
 			return $this->_setParent($Model, $Model->data[$Model->alias][$parent]);
 		}
 	}
 
 /**
+ * Set level for descendents.
+ *
+ * @param Model $Model Model using this behavior.
+ * @param int|string $id Record ID
+ * @return void
+ */
+	protected function _setChildrenLevel($Model, $id) {
+		$settings = $Model->Behaviors->Tree->settings[$Model->alias];
+		$primaryKey = $Model->primaryKey;
+		$depths = array($id => (int)$Model->data[$Model->alias][$settings['level']]);
+
+		$children = $Model->children(
+			$id,
+			false,
```

Comment:
I had to keep the api open side by side to ensure i got the order of params right :stuck_out_tongue: